### ISSUE TITLE
Used display type of the root element is not correct.

### ISSUE THREAD

**Issue by glennw** [servo/servo#7558]
See http://www.w3.org/TR/CSS2/visuren.html#dis-pos-flo

In particular:

"Otherwise, if the element is the root element, 'display' is set according to the table below, except that it is undefined in CSS 2.1 whether a specified value of 'list-item' becomes a computed value of 'block' or 'list-item'."

**Comment by aopicier** [comment]
I've looked at this problem a bit. Here are two documents which currently do not behave correctly (both mentioned in #6375):

```html
<html style="display:inline">
test
</html>
```
```html
<script>
    document.removeChild(document.documentElement);
    root = document.createElementNS(null, "html");
    root.appendChild(document.createTextNode("test"));
    document.appendChild(root);
</script>
```
In both cases, the computed display type of the root element is actually correctly, namely `block`. But the flow construction uses the `-servo-display-for-hypothetical-box` property instead of the `display` property in both cases, which has the value `inline`. Hence the flow construction tries to treat the root element as an inline element and fails to construct a flow. The `-servo-display-for-hypothetical-box` property was introduced to accommodate absolutely-positioned elements as in CSS 2.1 § 10.3.7, but its special meaning probably should not apply to the root element.

@pcwalton Does this make sense? I have prepared a possible fix but as I'm only starting to try to work on Servo I'm not at all sure whether it is in fact the way to go: https://github.com/aopicier/servo/commit/c2db10217c4d78a917dfb28b5d9f6bdb727a5556
An alternative would be to simply exclude the root element from the logic prioritizing `-servo-display-for-hypothetical-box` over the `display` property in construct.rs. However I'm not sure what would be the correct way to determine if we are dealing with the root element there. Also this would mean that a special treatment of the root element would have to be applied should `-servo-display-for-hypothetical-box` be used in other places in the future.

In any case I'd be very grateful for input.
